clients/da-vinci-client/src/test/java/com/linkedin/davinci/ingestion/IsolatedIngestionBackendTest.java

```diff
@@ -200,4 +203,58 @@ public void testIsolatedIngestionNotifierAsyncCompletionHandling() {
     backend.getIsolatedIngestionNotifier(ingestionNotifier).completed(topic, 1, 123L, "", Optional.empty());
     verify(backend, times(1)).getCompletionHandlingExecutor();
   }
+
+  @Test
+  public void testBackendCanMaintainMetadataCorrectlyForDroppingPartition() {
+    try (MainIngestionMonitorService monitorService = mock(MainIngestionMonitorService.class);
+        IsolatedIngestionBackend backend = mock(IsolatedIngestionBackend.class)) {
+      String topic = "testTopic_v1";
+      int partition = 0;
+      VeniceStoreVersionConfig storeVersionConfig = mock(VeniceStoreVersionConfig.class);
+      when(storeVersionConfig.getStoreVersionName()).thenReturn(topic);
+      MainIngestionRequestClient ingestionRequestClient = mock(MainIngestionRequestClient.class);
+      when(backend.getMainIngestionRequestClient()).thenReturn(ingestionRequestClient);
+      when(backend.getMainIngestionMonitorService()).thenReturn(monitorService);
+      Map<String, MainTopicIngestionStatus> topicIngestionStatusMap = new VeniceConcurrentHashMap<>();
+      MainTopicIngestionStatus topicIngestionStatus = new MainTopicIngestionStatus(topic);
+      topicIngestionStatus.setPartitionIngestionStatusToLocalIngestion(partition);
+      topicIngestionStatusMap.put(topic, topicIngestionStatus);
+      doCallRealMethod().when(monitorService).cleanupTopicPartitionState(topic, partition);
+      when(monitorService.getTopicPartitionIngestionStatus(topic, partition)).thenCallRealMethod();
+      when(monitorService.getTopicIngestionStatusMap()).thenReturn(topicIngestionStatusMap);
+      when(backend.isTopicPartitionHostedInMainProcess(anyString(), anyInt())).thenCallRealMethod();
+      when(backend.isTopicPartitionIngesting(anyString(), anyInt())).thenCallRealMethod();
+      doCallRealMethod().when(backend).dropStoragePartitionGracefully(any(), anyInt(), anyInt(), anyBoolean());
+      doCallRealMethod().when(backend).executeCommandWithRetry(anyString(), anyInt(), any(), any(), any());
+      when(monitorService.getTopicPartitionCount(topic)).thenReturn(2L);
+      ReentrantReadWriteLock readWriteLock = new ReentrantReadWriteLock();
+      when(monitorService.getForkProcessActionLock()).thenReturn(readWriteLock);
+
+      // Case 1
```

Comment:
nit: can we be verbose on what test case we want to exercise there? It can help us to figure out these test cases easily in the future.

Reply:
Thanks! I've updated the comment, please feel free to take a second look